### DISCUSSION TITLE
supplementary / illustrative variables implemented?

### DISCUSSION THREAD

**Discussion by mluerig** [MaxHalford/prince#183] [Q&A] [answered]
I was wondering whether supplementary variables are implemented (corresponding to `num.group.sup` argument in factominer)?

Also: I really appreciate your efforts with this - great initiative, great package, great name!

**Comment by MaxHalford** [comment]
Hey there @mluerig! I appreciate the kind words.

Yes, Prince supports supplementary variables:

```py
import prince

dataset = prince.datasets.load_energy_mix(year=2019, normalize=True)
pca = prince.PCA()
pca.fit(dataset, supplementary_columns=['hydro', 'wind', 'solar'])
pca.column_correlations
```

This is unit tested against FactoMineR, and should provide exactly the same results 👌

**Reply by mluerig** [reply]
shoot, forgot to say that I meant MFA - doesn't seem `mfa.fit `accepts supplementary_columns?

**Reply by MaxHalford** [reply, marked as answer]
Ah indeed, Prince does not yet support supplementary variables for MFA :/

**Comment by mluerig** [comment]
too bad. probably not high on your priority list, but looking at the code for PCA and seeing how MFA is implemented it might not be too hard to implement supporting variables there as well?

let me know if you have any plans to do so, otherwise I could give it a shot sometime next year...

**Reply by MaxHalford** [reply]
Indeed I don't think it should be too difficult. I will try to take a look over the Xmas holidays!

**Reply by MaxHalford** [reply]
Progress in #185!

**Comment by MaxHalford** [comment]
@mluerig I just released v0.15. I completely revamped MFA: it now properly inherits from PCA, and thus has all the latter's methods. Supplementary rows and supplementary groups are now supported.

Please let me know if you're looking for anything particular! I can strike the iron while it's hot.

Also, if I may ask, what are you using MFA for?

**Comment by mluerig** [comment]
so cool, thanks! will try to take a look in the next few days. I think supp. cols and rows were mostly what I was missing

I have been using MFA to analyze large datasets of phenotypic traits from various organisms, where the grouping either consituted different types of traits (shape vs color), or different parts of an organism (dorsum vs ventrum, thorax vs abdomen in insects). MFA seemed like a great way to construct such morphospaces (https://doi.org/10.1016/j.cub.2021.08.040) that account for group differences yet keep them comparable, because all subspaces are rotated in the same direction

**Comment by Nika250** [comment]
thanks for the information!
